src/library_pthread.js

```diff
@@ -211,11 +211,11 @@ var LibraryPThread = {
       // worker pool as an unused worker.
       worker.pthread_ptr = 0;
 
-#if ENVIRONMENT_MAY_BE_NODE
-      if (ENVIRONMENT_IS_NODE) {
-        // Once a pthread has finished and the worker becomes idle, mark it
-        // as weakly referenced so that its existence does not prevent Node.js
-        // from exiting.
+#if ENVIRONMENT_MAY_BE_NODE && PROXY_TO_PTHREAD
+      if (ENVIRONMENT_IS_NODE && worker.proxied_main) {
```

Comment:
How about if this read more like:

```
// Even threads that are strongly referenced, such as the proxied main thread
// or threads that are being kept alive long enough to proxy their error state
// should be unref() once that are returned to the poll
if (worker.strongly_referenced) {
   worker.unref();
   worker.is_strongly_referenced = false;
}
```

Then `_emscripten_thread_set_strongref` could set `is_strongly_referenced` to true and `PROZY_TO_PTHREAD` could explictly call `_emscripten_thread_set_strongref` on main's pthread instead of adding the magic extra argument to `__pthread_create_js`. ?

Reply:
Even better I think we can just unconditionally call `worker.unref();` here!   Since according the docs "If the worker is already unref()ed calling unref() again has no effect."  (https://nodejs.org/api/worker_threads.html#workerunref)

Reply:
Indeed, that extra argument added to `__pthread_create_js` was a bit of an weird way to identify the proxied main. I reverted those commits and instead added a comment to clarify that `.ref()`/`.unref()` is backed by a boolean.